compiler/src/dmd/globals.d

```diff
@@ -69,6 +69,13 @@ enum CppStdRevision : uint
     cpp20 = 2020_02,
 }
 
+enum DDocStructure : ubyte
+{
+    flat,           /// `module a.b.c;` => c.html
+    qualifiedNames, /// `module a.b.c;` => a_b_c.html
```

Comment:
It would be nice to have the option to use `.` as a separator:

```suggestion
    qualifiedNames, /// `module a.b.c;` => a.b.c.html
    qualifiedNamesUnderscore, /// `module a.b.c;` => a_b_c.html
```

Reply:
Perhaps the separator could be a separate option, e.g.:
`dmd -Dfs=name:'.'`, `dmd -Dfs=name:'_'`, `dmd -Dfs=name:'-'` (after the usual processing by the shell, the quotes don't reach argv)

Reply:
I like this idea although there are a lot of potential issues with it. Like, should the character be filtered? What if it’s an invalid character for a file name (like : on Windows), or a Unicode character?
Given that dots are used to separate domain names in URLs and file extensions in file names, I feel that using them by default might be an issue. However, since underscores can be used in module names they’re vulnerable to a similar issue. Maybe hyphens would be a better default? They can’t be used in module names, but can be used in file names on most modern systems and are completely fine in URLs as far as I know.